### DISCUSSION TITLE
Hackathon: 2024-03-05

### DISCUSSION THREAD

**Discussion by stefaniebutland** [earthaccess-dev/earthaccess#485] [Hack days]
Reporting out

Please create a new comment (at the very bottom, use the text box with a button that says "comment", not "reply") to write about what you did today!

![image](https://github.com/nsidc/earthaccess/assets/3608264/b50e7bbb-cf22-4cf8-afb6-86ce5081b465)

Please include links to GitHub issues or pull requests relevant to your work, and use those issues or pull requests as the "source of truth" for documenting the work; this makes future readers jobs easier :) It's OK to duplicate some information while writing a summary here!

Use the "reply" feature to have a discussion under any given comment.

**Comment by nikki-t** [comment]
Today, I started to design the modifications needed to return data on any services that are available for a collection per the mini roadmap comment found [here](https://github.com/nsidc/earthaccess/issues/447#issuecomment-1976907147).

I started to implement a method in the earthaccess.results.DataCollections class called 'services' that returns a list of services and information associated with each service.

I would like to query CMR to return information on a service such as (we can modify this later to better fit the plugins interface):
- URL
- Description
- Name
- ServiceOptions

I also wanted to make sure I am adding in new functionality that is consistent with current usage of earthaccess. I think it makes sense to implement a child class of ServiceQuery from the [Python CMR API](https://github.com/nasa/python_cmr/blob/1702100ab3f8dcb4843ab9da184612b372ec1854/cmr/queries.py#L948C7-L948C19) as is done by the classes in the search.py file. Then the above 'services' method can use that class to query CMR for service data. I think it may make the most sense to include the ServiceQuery class in the search.py file but am open to feedback!

**Comment by betolink** [comment]
@JessicaS11 and I tried to reproduce #453 with no luck on my side. We'll keep working on this tomorrow with a new Python environment and a fresh install. 

@battistowx started to work on implementing automatic handling of `.dodsrc` for Opendap, seems like it has to be duplicated in the user path and the local path.

```
HTTP.COOKIEJAR=~/.dod_cookies
HTTP.NETRC=1
HTTP.NETRCFILE=~/.netrc
```

@itcarroll wroked on #330 and seems like a fix could be ready this week. 

@Rhys-L also worked with us on these issues!

**Reply by Rhys-L** [reply]
Thanks for summarizing this @betolink! We also discussed adding some examples that help set up an ML pipeline to engage that community, and adding some docs/examples on working with downloaded hdf4 data for users who mostly work with older datasets. I'll start writing up examples using an hdf4 workflow since I'm currently working on a related project.

**Comment by danielfromearth** [comment]
@mfisher87, @chuckwondo, and I worked in a VS Code live session to add a skeleton of unit tests. The planned tests would mock parts of the Earthdata Login (EDL) and Common Metadata Repository (CMR) responses to check that the User Acceptance Testing (UAT) environment is being queried appropriately, for [PR-426](https://github.com/nsidc/earthaccess/pull/426). The work-in-progress was captured in [this commit](https://github.com/nsidc/earthaccess/pull/426/commits/79131584b034713f447934055b0d5ffc82e49b4a).
